### PR TITLE
🩹 Fix incorrect civicrm_preferences_date description for activityDate and searchDate

### PR DESCRIPTION
**Pull request by agileware-justin**
Overview
----------------------------------------
Date preferences, **searchDate** is not used for **Relationships**. The **Relationship** date fields all use the **activityDate** format.

This PR fixes the incorrect civicrm_preferences_date description for activityDate and searchDate.

Before
----------------------------------------

Help text is wrong, _not helpful_.

image

After
----------------------------------------

Help text is correct, _now helpful_.

image

Technical Details
----------------------------------------

Comments
----------------------------------------
SQL re-gen needed to be executed and is included.

[Agileware Ref](https://agileware.com.au): CIVICRM-2051

### CODE REVIEW

**Comment by seamuslee001** [comment]
Confirmed that is the case here https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/DAO/Relationship.php#L300 and https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/DAO/Relationship.php#L316 MOP

**Comment by agileware-justin** [comment]
How do we execute the SQL re-gen using Github actions?

**Comment by demeritcowboy** [comment]
Try something like this:
https://github.com/civicrm/civicrm-core/blob/1b878aa9c87c0762ea66b3385c673554175d6f28/CRM/Upgrade/Incremental/php/FiveTwentyNine.php#L73-L83

File: CRM/Upgrade/Incremental/sql/5.56.alpha1.mysql.tpl [hunk 1]
```diff
@@ -14,3 +14,9 @@ SET ca.state_province_id = @DHStateID
 WHERE csp.country_id = @indianCountryID AND csp.abbreviation IN ("DN", "DD");
 
 UPDATE civicrm_state_province SET is_active = 0 WHERE country_id = @indianCountryID AND abbreviation IN ("DN", "DD");
+
+-- Fix incorrect civicrm_preferences_date description for activityDate and searchDate
+
+UPDATE civicrm_preferences_date SET description = {localize}'{ts escape="sql"}Date for relationships. activities. contributions: receive, receipt, cancel. membership: join, start, renew. case: start, end.{/ts}'{/localize} WHERE civicrm_preferences_date.name = 'activityDate';
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/sql/5.56.alpha1.mysql.tpl:20]: @agileware-justin just remove `{localize}` and `{/localize}` here and you should be fine and in L22

Comment (agileware-justin) [reply CRM/Upgrade/Incremental/sql/5.56.alpha1.mysql.tpl]: Trial and error, waiting for the build to confirm.

**Comment by demeritcowboy** [comment]
Ah sorry I thought the localize meant it was a multilingual field. But the civicrm_generated seems to still have the old wording. I don't think you'd need to regenerate it completely for this - could just edit those two lines in it.

**Comment by agileware-justin** [comment]
@seamuslee001 you need to unmerge and then merge the latest commit which has the correctly generated sql/civicrm_generated.mysql (https://github.com/agileware/civicrm-core/suites/8748802103/artifacts/396235203)

**Comment by seamuslee001** [comment]
already got that covered in #24731 @agileware-justin

**Comment by agileware-justin** [comment]
Well, that's kinda odd.

**Pull request merged**